### PR TITLE
tests: add test for NIC switching from direct to virtual network type

### PR DESCRIPTION
**Pull request by KKoukiou**
The bugzilla reported issue was most likely fixed in 6fa84cf757.

Related to https://bugzilla.redhat.com/show_bug.cgi?id=1977669

### CODE REVIEW

File: test/check-machines-nics [hunk 1]
```diff
@@ -198,6 +198,14 @@ class TestMachinesNICs(VirtualMachinesCase):
     def testNICEdit(self):
         b = self.browser
 
+        def open(iface):
+            b.click("#vm-subVmTest1-network-{0}-edit-dialog".format(iface))
```
Comment (martinpitt) [inline test/check-machines-nics:202]: One of these days we'll teach you f-strings :grin:

File: test/check-machines-nics [hunk 2]
```diff
@@ -242,16 +248,20 @@ class TestMachinesNICs(VirtualMachinesCase):
...
-        b.wait_not_present("#vm-subVmTest1-network-2-edit-dialog-modal-window")
+        save(2)
 
         # The NICs table should display the permanent configuration which is different than the live which we just set
         b.wait_in_text("#vm-subVmTest1-network-2-source", "eth43")
 
+        # Change source type from direct to virtual network - https://bugzilla.redhat.com/show_bug.cgi?id=1977669
+        open(2)
+        b.select_from_dropdown("#vm-subVmTest1-network-2-select-type", "network")
+        save(2)
```
Comment (martinpitt) [inline test/check-machines-nics:263]: This is a pretty shallow test, as it only checks that the UI does not crash. Ideally it should test the actual effect in the virt XML, or in what the "attached networks" UI shows after that.

**Review by martinpitt — APPROVED** [review]
Thanks!

File: test/check-machines-nics (hunk 2 above)
Comment (KKoukiou) [reply test/check-machines-nics:263]: Previously it generated an error in the dialog making the dialog to actually not close, now the dialog closes. I can indeed check that the change took effect.

**Comment by KKoukiou** [comment]
Ok merging this and I am sending now follow up with extension of the testEditNIC, in order that it always verifies with the XML.

**Pull request merged**